.github/workflows/tsan.yaml

```diff
@@ -0,0 +1,43 @@
+name: tsan
+on: [push, pull_request]
```

Comment:
Do we really want tsan on every push?

Reply:
That's a good point, I just checked how the ignition packages are handling it, and it looks like only build tests are on `push`.

Reply:
https://github.com/open-rmf/rmf_simulation/pull/14/commits/ac54875018c132d37b83131f7b60f02dc9d67fb5